### PR TITLE
YAML checker for CI - ensure no unit testing cluster

### PR DESCRIPTION
**Pull request by cecille**
Adding a checker for YAML tests that reference the unit testing cluster. 

It looks like some of the tests were using this cluster to get around YAML limitations, but this can't be done in cert because it then requires devices to carry a unit testing cluster in their prod image.

This workflow can then serve as a landing place for other quirks we find as we're doing cert stuff.

NOTE that we do have two cert tests currently that do this, and they are NOT being fixed as part of this PR.

### CODE REVIEW

File: .github/workflows/cert_test_checks.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Checkout submodules & Bootstrap
+        uses: ./.github/actions/checkout-submodules-and-bootstrap
+        with:
+          platform: linux
+      - name: Run checks
+        run: |
+          ./scripts/run_in_build_env.sh "python3 scripts/tests/matter_yaml_linter.py"
```
Comment (andy31415) [inline .github/workflows/cert_test_checks.yaml:35]: please add trailing newline

**Review by andy31415 — DISMISSED** [review]

File: .github/workflows/cert_test_checks.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      - "src/app/tests/suites/certification/**"
+
+jobs:
+  check-certification-tests:
+    name: Check for common problems in certification tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Checkout submodules & Bootstrap
```
Comment (andy31415) [inline .github/workflows/cert_test_checks.yaml:29]: bootstrap and submodules seems painful. Would running on the `chip-build` image (which includes python) or using https://github.com/actions/setup-python be faster?

File: .github/workflows/cert_test_checks.yaml (hunk 1 above)
Comment (cecille) [reply .github/workflows/cert_test_checks.yaml]: done.

File: .github/workflows/cert_test_checks.yaml (hunk 2 above)
Comment (cecille) [reply .github/workflows/cert_test_checks.yaml]: done. yes.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by mkardous-silabs — APPROVED** [review]

**Pull request merged**